easy-ui-react/src/TextField/TextField.stories.tsx

```diff
@@ -0,0 +1,210 @@
+} from "../utilities/storybook";
+
+import { TextField, TextFieldProps } from "./TextField";
+
+type Story = StoryObj<typeof TextField>;
+
+const Template = (args: TextFieldProps) => <TextField {...args} />;
+
+const meta: Meta<typeof TextField> = {
+  title: "Components/TextField",
+  component: TextField,
+};
+
+export default meta;
+
+export const Standard: Story = {
+  render: Template.bind({}),
+  args: {
+    label: "Label",
+    placeholder: "Placeholder text",
+    helperText: "Optional helper text",
+  },
+  decorators: [InputDecorator],
+};
+
+export const Password: Story = {
+  render: Template.bind({}),
+  args: {
+    type: "password",
+    label: "Password",
+    placeholder: "Placeholder text",
+    helperText: "Optional helper text",
+  },
+  decorators: [InputDecorator],
+};
+
+export const Icon: Story = {
+  render: Template.bind({}),
+  args: {
+    label: "Label",
+    placeholder: "Placeholder text",
+    helperText: "Optional helper text",
+    iconAtStart: SearchIcon,
+  },
+  decorators: [InputDecorator],
+};
+
+export const SmallTextFields: Story = {
+  render: () => (
+    <>
+      <TextField
+        size="sm"
+        label="Label"
+        placeholder="Placeholder text"
+        helperText="Optional helper text"
+      />
+      <TextField
+        type="password"
+        size="sm"
+        label="Password"
+        placeholder="Placeholder text"
+        helperText="Optional helper text"
+      />
+      <TextField
+        size="sm"
+        label="Label"
+        placeholder="Placeholder text"
+        iconAtStart={SearchIcon}
+        helperText="Optional helper text"
+      />
+    </>
+  ),
+  decorators: [InputDecorator],
+};
+
+export const LargeTextFields: Story = {
+  render: () => (
+    <>
+      <TextField
+        size="lg"
+        label="Label"
+        placeholder="Placeholder text"
+        helperText="Optional helper text"
+      />
+      <TextField
+        type="password"
+        size="lg"
+        label="Password"
+        placeholder="Placeholder text"
+        helperText="Optional helper text"
+      />
+      <TextField
+        size="lg"
+        label="Label"
+        placeholder="Placeholder text"
+        iconAtStart={SearchIcon}
+        helperText="Optional helper text"
+      />
+    </>
+  ),
+  decorators: [InputDecorator],
+};
+
+export const LabelWithEmphasis: Story = {
+  render: Template.bind({}),
+  args: {
+    label: "Label",
+    emphasizedLabel: true,
+    helperText: "Optional helper text",
+  },
+  decorators: [InputDecorator],
+};
+
+export const Error: Story = {
+  render: Template.bind({}),
+  args: {
+    validationState: "invalid",
+    label: "Label",
+    errorText: "Optional error text",
+  },
+  decorators: [InputDecorator],
+};
+
+export const VisuallyHiddenLabel: Story = {
+  render: Template.bind({}),
+  args: {
+    label: "Label",
+    isLabelVisuallyHidden: true,
+  },
+  decorators: [InputDecorator],
+};
+
+export const Controls: Story = {
```

Comment:
couldn't get storybook to document the props correctly, hence the manual work for the controls story. it seems as though its having issues with the union between `TextFieldProps` and `AriaTextFieldProps`. when i remove `AriaTextFieldProps`, the props get documented correctly, but with the union (as we need it), it was only picking up the props unique to `TextFieldProps`.

Reply:
hm yeah that's unfortunate. this seems like a lot of effort to write/maintain